### PR TITLE
New package: igprof

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/igprof/package.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+class Igprof(CMakePackage):
+    """IgProf (the Ignominous Profiler) is a simple nice tool for measuring and
+       analysing application memory and performance characteristics.
+
+       IgProf requires no changes to the application or the build process. It
+       currently works on Linux (ia32, x86_64)."""
+
+    homepage = "https://igprof.org/"
+    url      = "https://github.com/igprof/igprof/archive/v5.9.16.tar.gz"
+
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/igprof/package.py:18]: Do you want to add yourself as a Spack package maintainer for this package?  That would give you a chance to be notified of and comment on any changes to this package.

**Review by tldahlgren — COMMENTED** [review]
Confirmed the specified version is available on the repository.

File: var/spack/repos/builtin/packages/igprof/package.py (hunk 1 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/igprof/package.py:18]: I would highly doubt any changes will happen (last commit was ages ago :) ). Anyway, the answer is "not yet".

File: var/spack/repos/builtin/packages/igprof/package.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    version('5.9.16', sha256='cc977466b310f47bbc2967a0bb6ecd49d7437089598346e3f1d8aaf9a7555d96')
+
+    depends_on('libunwind')
+
+    patch('igprof-5.9.16.patch', when='@5.9.16', level=0)
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/igprof/package.py:28]: Can you add a comment explaining why the patch is needed and stating its provenance?

**Review by alalazo — CHANGES_REQUESTED** [review]
LGTM. A minor request on the patch.

File: var/spack/repos/builtin/packages/igprof/package.py (hunk 2 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/igprof/package.py:28]: Added

**Review by alalazo — DISMISSED** [review]

**Review by alalazo — APPROVED** [review]

**Pull request merged**